### PR TITLE
Describe identifiers more clearly in help and documentation

### PR DESCRIPTION
**Pull request by Bharath-Ganesh**
The documentation for the input step used ID when taking about the id field.
I have refactored all the references of ID to id

LINK: https://issues.jenkins.io/browse/JENKINS-69889

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: README.md
Comment (MarkEWaite) [inline README.md:14]:
```suggestion
Input Example with `Custom identifier`:
Every input step has an unique identifier. It is used in the generated URL to proceed or abort.
```

File: README.md
Comment (MarkEWaite) [inline README.md:16]:
```suggestion
A specific identifier could be used, for example, to mechanically respond to the input from some external process/tool.
```

File: README.md
Comment (MarkEWaite) [inline README.md:26]: In this case, I think that many places in the Jenkins documentation and changelogs use the word "username" to refere to the identifier of a user.
```suggestion
Usernames and/or external group names of those permitted to respond to the input, separated by ','. Spaces will be trimmed automatically, so "Alice, Bob, Charles" is the same as "Alice,Bob,Charles".
```

File: README.md
Comment (MarkEWaite) [inline README.md:33]: Username rather than `id` (I believe, though I'm not thrilled with having two occurrences of `user` in the same sentence.
```suggestion
If specified, this is the name of the return value that will contain the username of the user that approves this input. The return value will be handled in a fashion similar to the parameters value.
```

File: src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-id.html
Comment (MarkEWaite) [inline src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-id.html:2]:
```suggestion
    Every <code>input</code> step has an unique identifier. It is used in the generated URL to proceed or abort.
```

File: src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-id.html
Comment (MarkEWaite) [inline src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-id.html:5]:
```suggestion
    A specific identifier could be used, for example, to mechanically respond to the input from some external process/tool.
```

**Review by MarkEWaite — COMMENTED** [review]
Thanks for the pull request and thanks for being involved in the Jenkins project.

I think that the guidance from @jtnord was that the acronym "ID" would be better expressed as the word "identifier" since it could be confused with many different forms of "ID".  The page linked by @daniel-beck shows many possible expansions of the acronym "ID".

I don't have a strong objection to the existing changes in this pull request, though I think the changes that I've proposed will better address the concerns expressed by @jtnord and the questions raised by @daniel-beck .

There is a reference to `Custom ID` that is not covered in this pull request.  I would not block the merge of this pull request based on that remaining reference, but I think it would be nice if this pull request or a future pull request also changed that reference.  The picture I took looks like this:

id-or-identifier

Again, this pull request is already a step forward.  I'm fine if it is merged as is or if it is merged with the changes that I suggested.

**Comment by MarkEWaite** [comment]
You may also want to rephrase the title of the pull request.

File: src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-id.html
Comment (Bharath-Ganesh) [reply src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-id.html:2]: done

File: README.md
Comment (Bharath-Ganesh) [reply README.md:33]: done

File: README.md
Comment (Bharath-Ganesh) [reply README.md:26]: done

**Review by MarkEWaite — APPROVED** [review]

**Comment by Bharath-Ganesh** [comment]
Thanks @MarkEWaite . Do I need to do anything else to get this merged?

**Comment by MarkEWaite** [comment]
> Thanks @MarkEWaite . Do I need to do anything else to get this merged?

Not as far as I can tell.  It will need to wait for a maintainer to review, approve, and merge it.

**Pull request merged**
